basil.js

```diff
@@ -893,10 +894,15 @@ pub.background = function (bgColor) {
   backgroundShape.strokeWeight = 0;
 
   bgLayer.locked = true;
-  if(pLayer.name != bgLayerName){
-    pub.layer(pLayer);
+  if (pLayer.name === bgLayerName) {
+    var docLayers = currentDoc().layers;
+    if (docLayers.length > 1) {
+      pub.layer(docLayers[0]);
+    }else{
+      pub.layer('output');
```

Comment:
I don't think we should pick an arbitrary name. Better let InDesign create the layer and name it according to the user's locale.
Also, I think it would be more natural not to select the foremost layer, but the layer right before the bgLayer (as this one was selected previously, so the selection does not jump away very far.

Reply:
Sounds good – adding a generic layer instead of one called 'output'.